src/main/java/org/folio/dew/batch/acquisitions/edifact/jobs/MapToEdifactOrdersTasklet.java

```diff
@@ -0,0 +1,113 @@
+    // Order filters
+    var workflowStatusFilter = "purchaseOrder.workflowStatus==Open"; // order status is Open
+    var vendorFilter = String.format(" AND purchaseOrder.vendor==%s", ediConfig.getVendorId()); // vendor id matches
+    var notManualFilter = " AND (cql.allRecords=1 NOT purchaseOrder.manualPo==true)"; // not a manual order
+
+    // Order line filters
+    var automaticExportFilter = " AND automaticExport==true"; // line with automatic export
+    var ediExportDateFilter = " AND (cql.allRecords=1 NOT lastEDIExportDate=\"\")"; // has not been exported yet
+    var acqMethodsFilter = fieldInListFilter("acquisitionMethod",
+      ediConfig.getEdiConfig().getDefaultAcquisitionMethods()); // acquisitionMethod in default list
+    String vendorAccountFilter = "";
+    if (Boolean.TRUE.equals(ediConfig.getIsDefaultConfig())) {
+      var configQuery = String.format("configName==%s_%s*", ExportType.EDIFACT_ORDERS_EXPORT, ediConfig.getVendorId());
+      var configs = dataExportSpringClient.getExportConfigs(configQuery);
+      if (configs.getTotalRecords() > 1) {
+        var accountNoSetForExclude = getAccountNoSet(configs);
+        vendorAccountFilter = fieldNotInListFilter("vendorDetail.vendorAccount", accountNoSetForExclude);
+      }
+    } else {
+      // vendorAccount in the config account number list
+      vendorAccountFilter = fieldInListFilter("vendorDetail.vendorAccount",
+        ediConfig.getEdiConfig().getAccountNoList());
+    }
+
+    var resultQuery = String.format("%s%s%s%s%s%s%s", workflowStatusFilter, vendorFilter, notManualFilter,
+      automaticExportFilter, ediExportDateFilter, acqMethodsFilter, vendorAccountFilter);
+    log.info("GET purchase order line query: {}", resultQuery);
+    return resultQuery;
+  }
+
+  private Set<String> getAccountNoSet(ExportConfigCollection configs) {
+    Set<String> accountNoSet = new HashSet<>();
+    for (ExportConfig exportConfig : configs.getConfigs()) {
+      EdiConfig ediConfig = exportConfig.getExportTypeSpecificParameters().getVendorEdiOrdersExportConfig().getEdiConfig();
+      if (Objects.nonNull(ediConfig)) {
+        List<String> currentAccountNoList = ediConfig.getAccountNoList();
+        if (CollectionUtils.isNotEmpty(currentAccountNoList)) {
+          accountNoSet.addAll(currentAccountNoList);
+        }
+      }
+    }
+    return accountNoSet;
+  }
+
+  private String fieldInListFilter(String fieldName, List<?> list) {
+    return String.format(" AND %s==%s", fieldName,
+      list.stream()
+        .map(item -> String.format("\"%s\"", item.toString()))
+        .collect(Collectors.joining(" OR ", "(", ")")));
+  }
+
+  private static String fieldNotInListFilter(String fieldName, Collection<?> list) {
+    return String.format(" AND cql.allRecords=1 NOT %s==%s", fieldName,
+      list.stream()
+        .map(item -> String.format("\"%s\"", item.toString()))
+        .collect(Collectors.joining(" OR ", "(", ")")));
```

Comment:
maybe better to move utils like https://github.com/folio-org/mod-orders/blob/e5b470a2370a84bcb9a677ef63683a3692baa4b9/src/main/java/org/folio/orders/utils/QueryUtils.java#L14-L13

Reply:
Reused existing util methods instead of these, both are removed